qt/scientific_interfaces/Inelastic/Processor/ElwinPresenter.h

```diff
@@ -68,28 +74,38 @@ class MANTIDQT_INELASTIC_DLL ElwinPresenter : public DataProcessor, public IElwi
   void handleRowModeChanged() override;
   void updateAvailableSpectra() override;
 
-  void setInputWorkspace(MatrixWorkspace_sptr inputWorkspace);
+  void setInputWorkspace(const MatrixWorkspace_sptr &inputWorkspace);
   virtual void setSelectedSpectrum(int spectrum);
   int getSelectedSpectrum() const;
-  MatrixWorkspace_sptr getInputWorkspace() const;
+  MatrixWorkspace_sptr getInputWorkspace() const override;
   MatrixWorkspace_sptr getPreviewPlotWorkspace();
   void setPreviewPlotWorkspace(const MatrixWorkspace_sptr &previewPlotWorkspace);
 
+  void removeWorkspace(std::string const &workspaceName) const override;
+
+  // Analysis Data Service Observer
+  void renameHandle(const std::string &wsName, const std::string &newName) override;
+  void deleteHandle(const std::string &wsName, const Workspace_sptr &ws) override;
+  void clearHandle() override;
+
 protected:
   void runComplete(bool error) override;
-  void newInputDataFromDialog();
   virtual void addDataToModel(MantidWidgets::IAddWorkspaceDialog const *dialog);
 
 private:
+  void updateInterface();
   void updateTableFromModel();
   void updateIntegrationRange();
 
   std::vector<std::string> getOutputWorkspaceNames();
   std::string getOutputBasename();
   bool checkForELTWorkspace();
-  void newPreviewFileSelected(const std::string &workspaceName, const std::string &filename);
-  void newPreviewWorkspaceSelected(const std::string &workspaceName);
-  WorkspaceID findWorkspaceID();
+  void newPreviewWorkspaceSelected(int index);
+  std::optional<WorkspaceID> findWorkspaceID(std::string const &name) const;
+
+  void handleRemoveEvent(Mantid::API::WorkspacePreDeleteNotification_ptr pNf);
+  void handleClearEvent(Mantid::API::ClearADSNotification_ptr pNf);
+  void handleRenameEvent(Mantid::API::WorkspaceRenameNotification_ptr pNf);
```

Comment:
Did you forget to remove these old handlers?

Reply:
Yes. thank you.